pkg/controller/plan/kubevirt.go

```diff
@@ -795,7 +795,14 @@ func (r *KubeVirt) getStorageProfileModes(storageName string) (accessModes []cor
 					claimPropertyVolumeMode = &filesystemMode
 				}
 				if *claimPropertyVolumeMode == *volumeMode {
-					accessModes = append(accessModes, claimProperty.AccessModes...)
+					for _, am := range claimProperty.AccessModes {
+						if am == core.ReadWriteMany {
+							accessModes = append(accessModes, am)
+							return
+						}
+					}
+					accessModes = append(accessModes, claimProperty.AccessModes[0])
```

Comment:
how about imitating the logic in CDI  - no need to take all accessModes but to take the ones of the first claim property

Reply:
I referred to [this](https://github.com/kubevirt/containerized-data-importer/blob/b98001422c95ebf54cb41d6bc81faab3f70bd6b1/pkg/controller/datavolume/util.go#L162-L167)

Reply:
Yes, definitely. The current code is unsafe and should change the array length.